python/imageutils.py

```diff
@@ -0,0 +1,481 @@
+
+                for y in range(0, num_y_iters):
+
+                    if y == num_y_iters - 1 and remainder_y != 0:
+                        first_dim = remainder_y
+                        is_rem_y = True
+
+                    x_start = x*third_dim
+                    x_end = (x+1)*third_dim
+
+                    z_start = z*second_dim
+                    z_end = (z+1)*second_dim
+
+                    y_start = y*first_dim
+                    y_end = (y+1)*first_dim
+
+                    split_array = self.proxy.dataobj[y_start:y_end, z_start:z_end, x_start:x_end]
+                    split_image = nib.Nifti1Image(split_array, self.proxy.header.get_best_affine())
+       
+                    imagepath = None
+                    
+                    #TODO: fix this so that position saved in image and not in filename
+                    #if the remaining number of voxels does not match the requested number of voxels, save the image with
+                    #with the given filename prefix and the suffix: _<x starting coordinate>_<y starting coordinate>_<z starting coordinate>__rem-<x lenght>-<y-length>-<z length>
+                    if is_rem_x or is_rem_y or is_rem_z:
+                        imagepath = '{0}/{1}_{2}_{3}_{4}__rem-{5}-{6}-{7}.nii.gz'.format(local_dir, filename_prefix, y_start, z_start, x_start, y_length, z_length, x_length) 
+                    else:
+                        imagepath = '{0}/{1}_{2}_{3}_{4}.nii.gz'.format(local_dir, filename_prefix, y_start, z_start, x_start)
+                    
+                    nib.save(split_image, imagepath)
+
+                    if copy_to_hdfs:
+                        self.utils.copy_to_hdfs(imagepath, ovrwrt=True, save_path_to_file=True)
+                    else:
+                        with open('{0}/legend.txt'.format(local_dir), 'a+') as im_legend:
+                            im_legend.write('{0}\n'.format(imagepath))
+
+                    is_rem_z = False
+        
+            is_rem_y = False
+
+    #TODO: Fix function so that it will work in HDFS
+    def cp_block_block(self, legend):
+        """ Reconstructs and image given a set of blocks. Filenames of blocks must contain their positioning in 3d in the reconstructed image
+        (will work for slices too as long as filenames are configured properly)
+
+        Keyword arguments:
+        legend                        : a legend containing the location of the blocks to use for reconstruction.
+
+
+        """
+
+        
+        rec_header = self.proxy.header
+        bytes_per_voxel = rec_header['bitpix']/8 
+        
+        rec_dims = self.proxy.header.get_data_shape()
+     
+        y_size = rec_dims[0]
+        z_size = rec_dims[1]
+        x_size = rec_dims[2]
+
+        #print self.filepath
+        with open(self.filepath, "r+b") as reconstructed:
+            with open(legend, 'r')  as legend:
+                prev_b_end = (0, 0, 0)
+                for block_name in legend:
```

Comment:
Right. I might just leave this for now and put it down as an issue. Since I'm creating a legend class, maybe i can configure it such that we don't really have to modify the code there, as the legend class will sort the blocks before returning the next block. How does that sound?